sql-plugin/src/main/311until340-all/scala/org/apache/spark/sql/hive/rapids/shims/GpuHiveTextFileFormat.scala

```diff
@@ -0,0 +1,178 @@
+import java.nio.charset.Charset
+import org.apache.hadoop.mapreduce.{Job, TaskAttemptContext}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.hive.rapids.GpuHiveTextFileUtils._
+import org.apache.spark.sql.types.{DataType, StructType}
+
+object GpuHiveTextFileFormat extends Logging {
+
+  private def checkIfEnabled(meta: GpuInsertIntoHiveTableMeta): Unit = {
+    if (!meta.conf.isHiveDelimitedTextEnabled) {
+      meta.willNotWorkOnGpu("Hive text I/O has been disabled. To enable this, " +
+        s"set ${RapidsConf.ENABLE_HIVE_TEXT} to true")
+    }
+    if (!meta.conf.isHiveDelimitedTextWriteEnabled) {
+      meta.willNotWorkOnGpu("writing Hive delimited text tables has been disabled, " +
+        s"to enable this, set ${RapidsConf.ENABLE_HIVE_TEXT_WRITE} to true")
+    }
+  }
+
+  def tagGpuSupport(meta: GpuInsertIntoHiveTableMeta)
+  : Option[ColumnarFileFormat] = {
+    checkIfEnabled(meta)
+
+    val insertCommand = meta.wrapped
+    val storage  = insertCommand.table.storage
+    if (storage.outputFormat.getOrElse("") != textOutputFormat) {
+      meta.willNotWorkOnGpu(s"unsupported output-format found: ${storage.outputFormat}, " +
+        s"only $textOutputFormat is currently supported")
+    }
+    if (storage.serde.getOrElse("") != lazySimpleSerDe) {
+      meta.willNotWorkOnGpu(s"unsupported serde found: ${storage.serde}, " +
+        s"only $lazySimpleSerDe is currently supported")
+    }
+
+    val serializationFormat = storage.properties.getOrElse(serializationKey, "1")
+    if (serializationFormat != ctrlASeparatedFormat) {
+      meta.willNotWorkOnGpu(s"unsupported serialization format found: " +
+        s"$serializationFormat, " +
+        s"only \'^A\' separated text output (i.e. serialization.format=1) " +
+        s"is currently supported")
+    }
+
+    val lineTerminator = storage.properties.getOrElse(lineDelimiterKey, newLine)
+    if (lineTerminator != newLine) {
+      meta.willNotWorkOnGpu(s"unsupported line terminator found: " +
+        s"$lineTerminator, " +
+        s"only newline (\'\\n\') separated text output is currently supported")
+    }
+
+    if (!storage.properties.getOrElse(escapeDelimiterKey, "").equals("")) {
+      meta.willNotWorkOnGpu("escapes are not currently supported")
+      // "serialization.escape.crlf" matters only if escapeDelimiterKey is set
+    }
+
+    val charset = Charset.forName(
+      storage.properties.getOrElse("serialization.encoding", "UTF-8"))
+    if (!charset.equals(Charsets.UTF_8)) {
+      meta.willNotWorkOnGpu("only UTF-8 is supported as the charset")
+    }
+
+    if (insertCommand.table.bucketSpec.isDefined) {
+      meta.willNotWorkOnGpu("bucketed tables are not supported")
+    }
+
+    if (insertCommand.conf.getConfString("hive.exec.compress.output", "false") == "true") {
```

Comment:
nit: what happens if this is set to TRUE? would it be better to fall back to the CPU unless we see exactly what we want instead of only falling back to the CPU if we see one exact thing that we don't want?

i.e.
```
if (insertCommand.conf.getConfString("hive.exec.compress.output", "false").toLower() != "false") {
```

Reply:
Argh. Thank you for pointing this out. Will fix shortly.

Reply:
Fixed.